sdk/identity/identity/src/credentials/managedIdentityCredential/arcMsi.ts

```diff
@@ -15,21 +21,22 @@ const logger = credentialLogger("ManagedIdentityCredential - ArcMSI");
 // Azure Arc MSI doesn't have a special expiresIn parser.
 const expiresInParser = undefined;
 
-function prepareRequestOptions(resource?: string): RequestPrepareOptions {
+function prepareRequestOptions(resource?: string): PipelineRequestOptions {
   const queryParameters: any = {
     resource,
     "api-version": azureArcAPIVersion
   };
 
+  const query = qs.stringify(queryParameters);
+
   return {
     // Should be similar to: http://localhost:40342/metadata/identity/oauth2/token
-    url: process.env.IDENTITY_ENDPOINT,
+    url: `${process.env.IDENTITY_ENDPOINT!}?${query}`,
     method: "GET",
-    queryParameters,
-    headers: {
+    headers: createHttpHeaders({
       Accept: "application/json",
-      Metadata: true
-    }
+      Metadata: "true"
```

Comment:
Is there any requirement that values for headers need to be string in the new Azure Core? Or is it that previously this would be stringified inside core-http?

Reply:
Azure Core only supports string values on the headers now, as far as typescript complained 🤔